Type `any` not handled

<!-- Please make sure to run `v up` before reporting any issues as it may have already been fixed.
     It's also advisable to update all relvant modules using `v outdated` and `v install` -->

**V version:** 0.1.29
**OS:** Linux

**What did you do?**
```
    struct My_type {
     fld any
    }
    
    fn (o My_type) str() string {
        return match o.fld {
            []any{} {"empty map"}
            else {"<<unhandled>>"}
        }
    }
    
    vrbl := My_type{ fld: []any{} }
    
    println("The value is '$vrbl'.")
```

**What did you expect to see?**
`The value is 'empty map'.`
 
**What did you see instead?**
```
$  ./v run /tmp/yna.v 
==================
/tmp/v/yna.tmp.c: In function ‘main__My_type_str’:
/tmp/v/yna.tmp.c:9397:57: error: ‘any’ undeclared (first use in this function)
  return (o.fld == __new_array_with_default(0, 0, sizeof(any), 0)) ? (tos_lit("any map")) : (tos_lit("<<unhandled>>"));
                                                         ^~~
/tmp/v/yna.tmp.c:9397:57: note: each undeclared identifier is reported only once for each function it appears in
/tmp/v/yna.tmp.c: In function ‘main__main’:
/tmp/v/yna.tmp.c:9401:84: error: ‘any’ undeclared (first use in this function)
  main__My_type vrbl = (main__My_type){.fld = __new_array_with_default(0, 0, sizeof(any), 0),};
```

--- a/vlib/v/checker/checker.v
+++ b/vlib/v/checker/checker.v
@@ -638,6 +638,9 @@ pub fn (mut c Checker) struct_decl(mut node ast.StructDecl) {
 			}
 		}
 		for i, field in node.fields {
+			if field.typ == ast.any_type {
+				c.error('struct field cannot be the `any` type, use generics instead', field.type_pos)
+			}
 			c.ensure_type_exists(field.typ, field.type_pos) or { return }
 			if field.typ.has_flag(.generic) {
 				has_generic_types = true

--- a/vlib/v/checker/tests/struct_field_with_any_type_err.out
+++ b/vlib/v/checker/tests/struct_field_with_any_type_err.out
@@ -1,0 +1,6 @@
+vlib/v/checker/tests/struct_field_with_any_type_err.vv:2:6: error: struct field cannot be the `any` type, use generics instead
+    1 | struct My_type {
+    2 |     fld any
+      |         ~~~
+    3 | }
+    4 |

--- a/vlib/v/checker/tests/struct_field_with_any_type_err.vv
+++ b/vlib/v/checker/tests/struct_field_with_any_type_err.vv
@@ -1,0 +1,8 @@
+struct My_type {
+	fld any
+}
+
+fn main() {
+	a := My_type{}
+	println(a)
+}

